github/Repository.py

```diff
@@ -1273,8 +1273,8 @@ def create_git_release(
         :rtype: :class:`github.GitRelease.GitRelease`
         """
         assert isinstance(tag, str), tag
-        assert isinstance(name, str), name
-        assert isinstance(message, str), message
+        assert generate_release_notes and is_optional(name, str) or isinstance(name, str), name
+        assert generate_release_notes and is_optional(message, str) or isinstance(message, str), message
         assert isinstance(draft, bool), draft
         assert isinstance(prerelease, bool), prerelease
         assert isinstance(generate_release_notes, bool), generate_release_notes
```

Comment:
I think if we swap the `or` elemens, this reads more clearly:
```suggestion
        assert isinstance(generate_release_notes, bool), generate_release_notes
        assert isinstance(name, str) or generate_release_notes and is_optional(name, str), name
        assert isinstance(message, str) or generate_release_notes and is_optional(message, str), message
        assert isinstance(draft, bool), draft
        assert isinstance(prerelease, bool), prerelease
```
And we need to assert `generate_release_notes` before using it as a `bool`.